### PR TITLE
fix: software artifacts are downloaded twice when the `update-list` feature is not available

### PR DESCRIPTION
**Pull request by Ruadhri17**
This small PR fixes issue with software artifacts being downloaded twice when `update-list` is not available. Extra validation step was introduced to check if artifact is downloaded.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
* #3062 
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by jarhodes314 — APPROVED** [review]

**Review by albinsuresh — COMMENTED** [review]
LGTM. Can a system test be added to fully confirm the effectiveness of the fix? You can just assert that the agent log contains only one "downloading" statement for that artefact.

**Review by didier-wenzek — COMMENTED** [review]
I would only log an `info!` telling that the agent installs the packages one after the other as [`update_list` is not supported for this type of packages ](https://github.com/thin-edge/thin-edge.io/blob/main/crates/core/plugin_sm/src/plugin.rs#L138)

File: crates/core/plugin_sm/src/plugin.rs [hunk 1]
```diff
@@ -134,7 +135,8 @@ pub trait Plugin {
         // Execute the updates
         if failed_updates.is_empty() {
             let outcome = self.update_list(&updates, command_log.as_deref_mut()).await;
-            if let Err(SoftwareError::UpdateListNotSupported(_)) = outcome {
+            if let Err(SoftwareError::UpdateListNotSupported(name)) = outcome {
+                info!("{}", SoftwareError::UpdateListNotSupported(name));
```
Comment (jarhodes314) [inline crates/core/plugin_sm/src/plugin.rs:139]:
```suggestion
            if let Err(err @ SoftwareError::UpdateListNotSupported(_)) = outcome {
                info!("{err}");
```

Comment (rina23q) [reply crates/core/plugin_sm/src/plugin.rs]: Thanks, applied

**Review by Bravo555 — APPROVED** [review]

**Review by didier-wenzek — APPROVED** [review]
Approved

**Pull request merged**